libyul/ObjectParser.cpp

```diff
@@ -104,11 +127,94 @@ shared_ptr<Block> ObjectParser::parseCode()
 	return parseBlock();
 }
 
+optional<ObjectParser::ReverseSourceNameMap> ObjectParser::tryGetSourceLocationMapping() const
+{
+	return tryGetSourceLocationMapping(m_scanner->currentCommentLiteral());
+}
+
+optional<ObjectParser::ReverseSourceNameMap> ObjectParser::tryGetSourceLocationMapping(std::string const& _text)
+{
+	// @use-src 0:"abc.sol" , "1:foo.sol" ,2:"bar.sol"
+	//
+	// UseSrcList := UseSrc (',' UseSrc)*
+	// UseSrc     := [0-9]+ ':' FileName
+	// FileName   := "(([^\"]|\.)*)"
+	printf("tryGetSourceLocationMapping: \"%s\"\n", _text.c_str());
+
+	// Matches some "@use-src TEXT".
+	static std::regex const lineRE = std::regex(
+		R"~~~((^|\s+)@use-src\s+(.*)$)~~~",
```

Comment:
Is it possible to use one regex for everything and iterate through the matches for the inner group of
`@use-src\s+(((\d+)\s*:\s*"(([^\"]|\.)*")*`?

Reply:
No, it's not.